File: src/main/java/com/jcloisterzone/io/message/ShepherdPlacementConfirmMessage.java
package com.jcloisterzone.io.message;

import com.jcloisterzone.io.MessageCommand;

@MessageCommand("SHEPHERD_CONFIRM")
public class ShepherdPlacementConfirmMessage extends AbstractMessage implements ReplayableMessage, RandomChangingMessage {

    private Double random;

    public ShepherdPlacementConfirmMessage() {
    }

    @Override
    public Double getRandom() {
        return random;
    }

    @Override
    public void setRandom(Double random) {
        this.random = random;
    }
}

File: src/main/ts/com/jcloisterzone/io/message/ShepherdPlacementConfirmMessage.ts
import { AbstractMessage } from "./AbstractMessage.js";
import type { RandomChangingMessage } from "./RandomChangingMessage.js";
import type { ReplayableMessage } from "./ReplayableMessage.js";

/** SHEPHERD_CONFIRM — confirm a just-placed shepherd (draws the first sheep token). */
export class ShepherdPlacementConfirmMessage
  extends AbstractMessage
  implements ReplayableMessage, RandomChangingMessage
{
  static readonly command = "SHEPHERD_CONFIRM";

  private random: number | null = null;

  getRandom(): number | null {
    return this.random;
  }
  setRandom(random: number | null): void {
    this.random = random;
  }
}
